pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import {RouteComponentProps} from "react-router";
+import {createSelector} from "reselect";
+import {databaseNameAttr} from "src/util/constants";
+import {getMatchParamByName} from "src/util/query";
+import {AppState} from "../store";
+import {Dispatch} from "redux";
+import {actions as databaseDetailsActions} from "../store/databaseDetails";
+import {actions as localStorageActions, LocalStorageKeys} from "../store/localStorage";
+import {actions as tableDetailsActions} from "../store/databaseTableDetails";
+import {
+  DatabaseDetailsPageActions,
+  DatabaseDetailsPageData,
+  DatabaseDetailsPageDataTable,
+  ViewMode
+} from "./databaseDetailsPage";
+import {actions as analyticsActions} from "../store/analytics";
+import {Filters} from "../queryFilter";
+import {nodeRegionsByIDSelector} from "../store/nodes";
+import {selectIsTenant} from "../store/uiConfig";
+import {
+  selectDatabaseDetails,
+  selectDatabaseDetailsGrantsSortSetting,
+  selectDatabaseDetailsTablesFiltersSetting,
+  selectDatabaseDetailsTablesSearchSetting,
+  selectDatabaseDetailsTablesSortSetting,
+  selectDatabaseDetailsViewModeSetting
+} from "../store/databaseDetails/databaseDetails.selectors";
+import {
+  combineLoadingErrors,
+  generateTableID,
+  getNodesByRegionString,
+  normalizePrivileges,
+  normalizeRoles
+} from "../util";
+import {selectTableDetails} from "../store/databaseTableDetails/tableDetails.selector";
+
+const selectTables = (state: AppState, props: RouteComponentProps): DatabaseDetailsPageDataTable[] => {
```

Comment:
Is all this selector logic now being duplicated across db-console and cluster-ui?

Reply:
Refactored to use common selector logic between database components in db-console & cluster-ui. See `databasesCommon.selectors.ts`